### ISSUE TITLE
Layout transition life-cycle bug

### ISSUE THREAD

**Issue by arlm** [diogobernardino/WilliamChart#237]
Due to changes on life-cycle during layout transition, the component fails with NPE on the exit transition.

Pre-Oreo devices first detach the view and then tries to create an image of the view to animate the layout transition. Oreo devices first do the image and then detaches.

Due to this difference it only works properly on Android 8+ devices when using exit layout transitions. The error is that we do clean the style on detach and then it tries to paint the image. we fail to have a proper barPaint object by then.

```
05-30 15:13:23.762 24162-24162/org.providentiam.jejuniisdiebus E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.providentiam.jejuniisdiebus, PID: 24162
    java.lang.NullPointerException: Attempt to invoke virtual method 'void android.graphics.Paint.setColor(int)' on a null object reference
        at com.db.chart.view.BarChartView.onDrawChart(BarChartView.java:81)
        at com.db.chart.view.ChartView.onDraw(ChartView.java:363)
        at android.view.View.draw(View.java:17185)
        at android.view.View.buildDrawingCacheImpl(View.java:16474)
        at android.view.View.buildDrawingCache(View.java:16335)
        at android.view.View.draw(View.java:16943)
        at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
        at android.support.constraint.ConstraintLayout.dispatchDraw(ConstraintLayout.java:1963)
        at android.view.View.draw(View.java:17091)
        at android.view.ViewGroup.drawChild(ViewGroup.java:3727)
        at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3513)
        at android.view.View.draw(View.java:17188)
        at android.transition.TransitionUtils.createViewBitmap(TransitionUtils.java:168)
        at android.transition.TransitionUtils.copyViewImage(TransitionUtils.java:102)
        at android.transition.Visibility.onDisappear(Visibility.java:380)
        at android.transition.Visibility.createAnimator(Visibility.java:249)
        at android.transition.Transition.createAnimators(Transition.java:732)
        at android.transition.TransitionSet.createAnimators(TransitionSet.java:396)
        at android.transition.TransitionSet.createAnimators(TransitionSet.java:396)
        at android.transition.Transition.playTransition(Transition.java:1777)
        at android.transition.TransitionManager$MultiListener.onPreDraw(TransitionManager.java:306)
        at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:944)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2205)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1254)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6337)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:874)
        at android.view.Choreographer.doCallbacks(Choreographer.java:686)
        at android.view.Choreographer.doFrame(Choreographer.java:621)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:860)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

The error happens on https://github.com/diogobernardino/WilliamChart/blob/48091db7321ce935c37dedad44a62902126c08cb/williamchart/src/main/java/com/db/chart/view/BarChartView.java#L81

You can reproduce using the code and releases on this repo: https://github.com/Vita-Providentiam/Jejuniis-Android/releases/tag/v0.1.0

**Comment by diogobernardino** [comment]
Thanks for the analysis. I will consider it for williamchart v3, as I'm currently re-writing it from scratch in Kotlin. I will leave the issue open in case anyone wants to send a PR.

**Comment by arlm** [comment]
Cool. How is the v3 rewriting going? Are there some kind of list of tasks or tests to fulfil so that others could help?

**Comment by diogobernardino** [comment]
Better late than never, I just published v3.0. :) It's time to discontinue 2.x.

**Comment by arlm** [comment]
Cool.
